src/library_idbfs.js

```diff
@@ -23,8 +23,78 @@ addToLibrary({
+      }
+      function startPersist() {
+        mount.idbPersistState = 'idb'; // Mark that we are currently running a sync operation
+        IDBFS.syncfs(mount, /*populate:*/false, onPersistComplete);
+      }
+
+      if (!mount.idbPersistState) {
+        // Programs typically write/copy/move multiple files in the in-memory
+        // filesystem within a single app frame, so when a filesystem sync
+        // command is triggered, do not start it immediately, but only after
+        // the current frame is finished. This way all the modified files
+        // inside the main loop tick will be batched up to the same sync.
+        mount.idbPersistState = setTimeout(startPersist, 0);
+      } else if (mount.idbPersistState === 'idb') {
+        // There is an active IndexedDB sync operation in-flight, but we now
+        // have accumulated more files to sync. We should therefore queue up
+        // a new sync after the current one finishes so that all writes
+        // will be properly persisted.
+        mount.idbPersistState = 'again';
+      }
+    },
+
+    mount: (...args) => { 
+      var mnt = MEMFS.mount(...args);
+      mnt.idbPersistState = 0; // IndexedDB sync starts in idle state
+      var memfs_node_ops = mnt.node_ops;
+      mnt.node_ops = Object.assign({}, mnt.node_ops); // Clone node_ops to inject write tracking
+      mnt.node_ops.mknod = (parent, name, mode, dev) => {
+        var node = memfs_node_ops.mknod(parent, name, mode, dev);
+        node.node_ops = mnt.node_ops; // Propagate injected node_ops to the newly created child node
+        node.idbfs_mount = mnt.mount; // Remember for each IDBFS node which IDBFS mount point they came from so we know which mount to persist on modification.
+        node.memfs_stream_ops = node.stream_ops; // Remember original MEMFS stream_ops for this node
+        node.stream_ops = Object.assign({}, node.stream_ops); // Clone stream_ops to inject write tracking
+
+        // Track all file writes
+        node.stream_ops.write = (stream, buffer, offset, length, position, canOwn) => {
+          stream.node.isModified = true; // This file has been modified, we must persist IndexedDB when this file closes
+          return node.memfs_stream_ops.write(stream, buffer, offset, length, position, canOwn);
+        };
+
+        // Persist IndexedDB on file close
+        node.stream_ops.close = (stream) => {
+          var n = stream.node;
+          if (n.isModified) {
+            IDBFS.queuePersist(n.idbfs_mount);
+            n.isModified = false;
+          }
+          if (n.memfs_stream_ops.close) return n.memfs_stream_ops.close(stream);
+        };
+
+        return node;
+      };
+      // Also kick off persisting the filesystem on other operations that modify the filesystem.
+      mnt.node_ops.mkdir   = (...args) => { var ret = memfs_node_ops.mkdir(...args);   IDBFS.queuePersist(mnt.mount); return ret; };
```

Comment:
Can you write these as comma expressions to avoid the return statement and curly braces?

```
 mnt.node_ops.mkdir   = (...args) =>  IDBFS.queuePersist(mnt.mount), memfs_node_ops.mkdir(...args);
```

Reply:
Hmm, does that parse?

```js
function f() { return 1; }
function g() { return 2; }
var foo = (...args) => f(), g();
```
gives a syntax error, though
```js
function f() { return 1; }
function g() { return 2; }
var foo = (...args) => { return f(), g() };
```
works.

Reply:
Oh interesting.  Looks like simple comma expression work: `var foo = (...args) => f, g;`.

More complex ones require braces I guess: `var foo = (...args) => (f(), g());`